expirationwatch/expiration_watcher_test.go

```diff
@@ -0,0 +1,160 @@
+package expirationwatch
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPrunesExpiredItems(t *testing.T) {
+	watcher := New(0)
+
+	current := time.Now().Unix()
+	expiryEntryOne := ExpiredItem{
+		ExpirationTimeSeconds: current - 3,
+		ID:                    "0x8e209dda7e515025d0c34aa61a0d1156a631248a4318576a2ce0fb408d97385e",
+	}
+	watcher.Add(expiryEntryOne.ExpirationTimeSeconds, expiryEntryOne.ID)
+
+	expiryEntryTwo := ExpiredItem{
+		ExpirationTimeSeconds: current - 1,
+		ID:                    "0x12ab7edd34515025d0c34aa61a0d1156a631248a4318576a2ce0fb408d3bee521",
+	}
+	watcher.Add(expiryEntryTwo.ExpirationTimeSeconds, expiryEntryTwo.ID)
+
+	pruned := watcher.prune()
+	assert.Len(t, pruned, 2, "Pruned the expired item")
+	assert.Equal(t, expiryEntryOne, pruned[0])
+	assert.Equal(t, expiryEntryTwo, pruned[1])
+}
+
+func TestPrunesTwoExpiredItemsWithSameExpiration(t *testing.T) {
+	watcher := New(0)
+
+	current := time.Now().Unix()
+	expiration := current - 3
+	expiryEntryOne := ExpiredItem{
+		ExpirationTimeSeconds: expiration,
+		ID:                    "0x8e209dda7e515025d0c34aa61a0d1156a631248a4318576a2ce0fb408d97385e",
+	}
+	watcher.Add(expiryEntryOne.ExpirationTimeSeconds, expiryEntryOne.ID)
+
+	expiryEntryTwo := ExpiredItem{
+		ExpirationTimeSeconds: expiration,
+		ID:                    "0x12ab7edd34515025d0c34aa61a0d1156a631248a4318576a2ce0fb408d3bee521",
+	}
+	watcher.Add(expiryEntryTwo.ExpirationTimeSeconds, expiryEntryTwo.ID)
+
+	pruned := watcher.prune()
+	assert.Len(t, pruned, 2, "Pruned the expired item")
```

Comment:
This is a nitpick and you don't have to change it right now, but it would be easier to read test failures if the custom messages passed to `assert` or `require` more closely matched the grammar and structure of the default messages. The default messages describe _why_ the test failed or _how_ the expected result did not match the actual result, not _what_ you were testing. (e.g. "The two words should be the same" from the [docs page](https://godoc.org/github.com/stretchr/testify/assert) or "[1, 2, 3] should have 4 items but had 3" from [`assert.Len`](https://github.com/stretchr/testify/blob/34c6fa2dc70986bccbbffcc6130f6920a924b075/assert/assertions.go#L581)).

So instead of "Pruned the expired item" it should be something like "The expired item should be pruned but it was not". As it is currently written, the error message could be interpreted to mean that the expected behavior was that the item _should not_ be pruned but it actually was.